src/nvim/testdir/test_search.vim

```diff
@@ -346,25 +346,90 @@ func Test_searchc()
   bw!
 endfunc
 
-func Test_search_cmdline3()
+func Cmdline3_prep()
   throw 'skipped: Nvim does not support test_override()'
-  if !exists('+incsearch')
-    return
-  endif
   " need to disable char_avail,
   " so that expansion of commandline works
   call test_override("char_avail", 1)
   new
   call setline(1, ['  1', '  2 the~e', '  3 the theother'])
   set incsearch
+endfunc
+
+func Incsearch_cleanup()
+  throw 'skipped: Nvim does not support test_override()'
+  set noincsearch
+  call test_override("char_avail", 0)
+  bw!
+endfunc
+
+func Test_search_cmdline3()
+  throw 'skipped: Nvim does not support test_override()'
+  if !exists('+incsearch')
+    return
+  endif
+  call Cmdline3_prep()
   1
   " first match
   call feedkeys("/the\<c-l>\<cr>", 'tx')
   call assert_equal('  2 the~e', getline('.'))
-  " clean up
-  set noincsearch
-  call test_override("char_avail", 0)
-  bw!
+
+  call Incsearch_cleanup()
+endfunc
+
+func Test_search_cmdline3s()
+  throw 'skipped: Nvim does not support test_override()'
+  if !exists('+incsearch')
+    return
+  endif
+  call Cmdline3_prep()
+  1
+  call feedkeys(":%s/the\<c-l>/xxx\<cr>", 'tx')
+  call assert_equal('  2 xxxe', getline('.'))
+
```

Comment:
Missing tests from patch 8.1.0274.

Reply:
For this hunk:
```
diff --git a/src/nvim/testdir/test_search.vim b/src/nvim/testdir/test_search.vim
index 71de172dc..7e8ddbe64 100644
--- a/src/nvim/testdir/test_search.vim
+++ b/src/nvim/testdir/test_search.vim
@@ -516,11 +522,11 @@ func Test_search_cmdline7()
   endif
   " need to disable char_avail,
   " so that expansion of commandline works
   call test_override("char_avail", 1)
   new
-  let @/='b'
+  let @/ = 'b'
   call setline(1, [' bbvimb', ''])
   set incsearch
   " first match
   norm! gg0
   " moves to next match of previous search pattern, just like /<cr>
/
```
It changes the line on `Test_search_cmdline7`, which was not ported from mainline. It was introduced in patch 8.0.1250. Should I port the whole test? Should it be part of this changeset or should I make another PR?

Reply:
Port the whole test. It was omitted because of `test_override` but that makes it harder to port patches.